osbs/api.py

```diff
@@ -351,25 +352,30 @@ def create_prod_build(self, git_uri, git_ref,
         :param git_ref: str, reference to commit
         :param git_branch: str, branch name (may be None)
         :param user: str, user name
-        :param component: str, component name
-        :param target: str, koji target (may be None)
+        :param component: str, not used anymore
+        :param target: str, koji target
         :param architecture: str, build architecture
         :param yum_repourls: list, URLs for yum repos
         :param koji_task_id: int, koji task ID requesting build
         :param scratch: bool, this is a scratch build
         :return: BuildResponse instance
         """
+
+        def get_required_label(labels, names):
+            for label_name in names:
+                if label_name in labels:
+                    return labels[label_name]
+            
+            raise OsbsValidationException("required label '{name}' missing "
+                                          "from Dockerfile"
+                                          .format(name=label_name))
+
         df_parser = utils.get_df_parser(git_uri, git_ref, git_branch=git_branch)
         build_request = self.get_build_request()
         labels = df_parser.labels
-        for name_label_name in ['name', 'Name']:
-            if name_label_name in labels:
-                name_label = labels[name_label_name]
-                break
-        else:
-            raise OsbsValidationException("required label '{name}' missing "
-                                          "from Dockerfile"
-                                          .format(name=name_label_name))
+
+        name_label = get_required_label(labels, ('name', 'Name'))
```

Comment:
I think we want to the order to be old-style, then new-style. i.e., `('Name', 'name')`
This will match the order that is used for the component label as well.
It'll also report the new-style label in the error, which is preferred.